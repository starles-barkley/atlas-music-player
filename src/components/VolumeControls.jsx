const VolumeControls = () => {
  return (
    <div className="flex items-center space-x-2">
      <label htmlFor="volume" className="sr-only">
        Volume
      </label>
      <input
        id="volume"
        type="range"
        min="0"
        max="100"
        className="w-full h-2 bg-gray-200 rounded-lg appearance-none cursor-pointer focus:outline-none focus:ring-2 focus:ring-blue-500"
      />
    </div>
  );
};

export default VolumeControls;
