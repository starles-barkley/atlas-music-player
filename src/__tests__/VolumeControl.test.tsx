import { render, screen, fireEvent, cleanup } from "@testing-library/react";
import { expect, test, vi, afterEach } from "vitest";
import VolumeControls from "../components/VolumeControls";

afterEach(() => {
    cleanup();
});

test("default value of volume is 50", () => {
    const defaultVolume = 50;
    const setVolume = vi.fn();

    const { container } = render(<VolumeControls volume={defaultVolume} setVolume={setVolume} />);

    const slider = screen.getByRole("slider");
    expect(slider).toBeInTheDocument();
    expect(slider).toHaveValue("50");

    expect(container).toMatchSnapshot();
});

test("slider change updates volume and calls setVolume", () => {
    const setVolume = vi.fn();
    let initialVolume = 50;

    const mockSetVolume = (value: number) => {
        initialVolume = value;
        setVolume(value);
    };

    const { container, rerender } = render(<VolumeControls volume={initialVolume} setVolume={mockSetVolume} />);
    const sliders = screen.getAllByTestId("volume-slider");

    expect(sliders).toHaveLength(1);
    const slider = sliders[0];

    fireEvent.change(slider, { target: { value: 75 } });

    expect(setVolume).toHaveBeenCalledWith(75);
    rerender(<VolumeControls volume={initialVolume} setVolume={mockSetVolume} />);
    expect(slider).toHaveValue("75");
    expect(container).toMatchSnapshot();
});