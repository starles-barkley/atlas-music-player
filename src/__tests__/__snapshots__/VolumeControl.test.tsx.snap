// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`VolumeControls > matches snapshot 1`] = `
<div>
  <div
    class="flex items-center gap-4 rounded-lg bg-surface p-6 shadow-lg drop-shadow-md"
  >
    <svg
      aria-label="volume"
      class="lucide lucide-audio-lines text-secondary"
      fill="none"
      height="24"
      role="img"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M2 10v3"
      />
      <path
        d="M6 6v11"
      />
      <path
        d="M10 3v18"
      />
      <path
        d="M14 8v7"
      />
      <path
        d="M18 5v13"
      />
      <path
        d="M22 10v3"
      />
    </svg>
    <input
      aria-label="Volume control"
      class="h-2 w-full cursor-pointer appearance-none rounded-lg accent-primary"
      max="100"
      min="0"
      type="range"
      value="50"
    />
  </div>
</div>
`;
